src/api/participants/handler.js

```diff
@@ -93,18 +93,30 @@ const create = async function (request, h) {
     const settlementModels = allSettlementModels.filter(model => model.currencyId === request.payload.currency)
     if (Array.isArray(settlementModels) && settlementModels.length > 0) {
       for (const settlementModel of settlementModels) {
-        const participantCurrencyId1 = await ParticipantService.createParticipantCurrency(participant.participantId, request.payload.currency, settlementModel.ledgerAccountTypeId, false)
-        const participantCurrencyId2 = await ParticipantService.createParticipantCurrency(participant.participantId, request.payload.currency, settlementModel.settlementAccountTypeId, false)
+        const [participantCurrencyId1, participantCurrencyId2] = await Promise.all([
```

Comment:
I find this whole section from 96 -> 119 impossible to read. 

1. We are nesting 3 else blocks deep - why not `return` early on line `104` before the else block?
2. There is a lot of copying and pasting going on - line 96 -> 103 is almost the same as 108 -> 116. Let's make some generic functions to do this work and make the code more manageable
3. The error case where no default settlement model is set on line `118` is hidden at the bottom with a bunch of else statements. As a reader, there is no easy way for me to understand the conditions that would lead to triggering this error.